api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,248 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.ValidationException;
+import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * A utility class that evaluates the concept ranges 
+ * 
+ * @since 2.7.0
+ */
+public class ConceptReferenceRangeUtility {
+	
+	private static final Logger logger = LoggerFactory.getLogger(ConceptReferenceRangeUtility.class);
+	
+	public ConceptReferenceRangeUtility() {
+	}
+	
+	/**
+	 * This method evaluates the given criteria against the provided person/patient.
+	 *
+	 * @param criteria the criteria string to evaluate e.g. "$fn.getAge($patient, 'YEARS') > 1 && $fn.getAge($patient, 'YEARS') < 10"
+	 * @param person person object containing attributes to be used in the criteria
+	 *                  
+	 * @return true if the criteria evaluates to true, false otherwise
+	 * 
+	 * @since 2.7.0
+	 */
+	public boolean evaluateCriteria(String criteria, Person person) {
+		if (person == null) {
+			throw new ValidationException("Failed to validate with reason: patient is null");
+		}
+		
+		if (criteria == null || criteria.isEmpty()) {
+			throw new IllegalArgumentException("Failed to validate with reason: criteria required");
+		}
+		
+		criteria = improveCriteria(criteria);
+		
+		VelocityContext velocityContext = new VelocityContext();
+		velocityContext.put("fn", new ConceptReferenceRangeUtility());
+		
+		velocityContext.put("patient", person);
+		
+		VelocityEngine velocityEngine = new VelocityEngine();
+		
+		StringWriter writer = new StringWriter();
+		String wrappedCriteria = "#set( $criteria = " + criteria + " )$criteria";
+		velocityEngine.evaluate(velocityContext, writer, ConceptReferenceRangeUtility.class.getName(), wrappedCriteria);
+		
+		String evaluatedCriteria = writer.toString();
+		
+		return Boolean.parseBoolean(evaluatedCriteria);
+	}
+	
+	/**
+	 * Method is used to improve/build criteria so that a user won't have to add "patient" arg to the get age methods
+	 *
+	 * @return criteria
+	 *
+	 * @since 2.7.0
+	 */
+	private String improveCriteria(String criteria) {
+		if (criteria.contains("$fn.getAge()")) {
+			criteria = criteria.replace("$fn.getAge()", "$fn.getAge($patient)");
```

Comment:
Let us completely get rid of the replacements. It is enough to just implement `$fn.getAge()`,  `$fn.getAgeInMonths()`, `$fn.getAgeInWeeks()`, and `$fn.getAgeInDays()`

Reply:
Don't we need patient to get patient age? Another way to avoid this would be to set patient in a gloabal variable. However, when we do so, when velocity callge `getAge()` method, patient is null (probably because the context is different). Any suggestion you had in mind?

Reply:
Could you remind me why we did not add these methods to the person class?

Reply:
@ibacher  suggested to move these methods to Utility class

Reply:
Do you remember the reason he gave?

Reply:
Basically, I was thinking that we'd either end up with 4-8 new functions on Person (`getAgeInDays()`, `getAgeInMonths()`, maybe `getAgeInHours()` and `getAgeInWeeks()` (since reference ranges might be written in those) and then a variant that takes the date as the date to calculate the age at for each) or we could have a single function like:

```
getAgeInUnits(HOURS / DAYS / WEEKS / MONTHS / YEARS);
```

I was sort of figuring the person was part of the global namespace already. Alternatively, we could add a similar function to Person, I suppose.